seqr/views/apis/anvil_workspace_api.py

```diff
@@ -21,14 +21,21 @@
 from seqr.utils.communication_utils import send_html_email
 from seqr.utils.file_utils import does_file_exist
 from seqr.utils.logging_utils import SeqrLogger
-from seqr.views.utils.permissions_utils import is_anvil_authenticated, check_workspace_perm
-from settings import BASE_URL, GOOGLE_LOGIN_REQUIRED_URL
+from seqr.views.utils.permissions_utils import is_anvil_authenticated, check_workspace_perm, login_and_policies_required
+from settings import BASE_URL, GOOGLE_LOGIN_REQUIRED_URL, POLICY_REQUIRED_URL, API_POLICY_REQUIRED_URL
 
 logger = SeqrLogger(__name__)
 
 anvil_auth_required = user_passes_test(is_anvil_authenticated, login_url=GOOGLE_LOGIN_REQUIRED_URL)
 
-@anvil_auth_required
+def anvil_auth_and_policies_required(wrapped_func=None, policy_url=API_POLICY_REQUIRED_URL):
+    def decorator(view_func):
+        return login_and_policies_required(anvil_auth_required(view_func), login_url=GOOGLE_LOGIN_REQUIRED_URL, policy_url=policy_url)
+    if wrapped_func:
+        return decorator(wrapped_func)
+    return decorator
+
+@anvil_auth_and_policies_required(policy_url=POLICY_REQUIRED_URL)
```

Comment:
Does it work with two decorators rather than one cascading decorator? If yes, maybe two decorators could be more flexible.

Reply:
thats really inconsistent with our style for decorators everywhere else, and is less readable. It makes it seem like the decorators do different things instead of the same thing